v2python/tuning_database.py

```diff
@@ -1,284 +1,69 @@
-                # assert key_detected is not None, f'Functional(s) {mfsel.argument_names} are not found in the database'
-            self._index_matching_keys.append(key_detected)
-            self._fsel_positions.append(fsel.meta.first_apperance)
-        # print(f'{self._index_matching_keys=}')
-
-    def extract_keys_from_json(self, ti):
-        keys = [ti['inputs'].get(k, None) for k in self._index_matching_keys]
-        def convert(value):
-            if isinstance(value, str) and value.startswith('torch.'):
-                if value == 'torch.float16':
-                    return '*fp16:16'
-                elif value == 'torch.bfloat16':
-                    return '*bf16:16'
-                else:
-                    assert False, f'Unknown datatype {value}'
-            return value
-        return tuple(map(convert, keys))
-
-    def extract_keys_from_fsels(self, fsels, use_fallback_for_partially_tuned=False):
-        keys = {}
-        fallback_applied = []
-        # print(f'{len(self._fsel_positions)=}')
-        for fsel in fsels:
-            try:
-                # print(f'{fsel=}')
-                # print(f'extract_keys_from_fsels {fsel.argument_names} {fsel.meta.first_apperance}')
-                offset = self._fsel_positions.index(fsel.meta.first_apperance)
-                if use_fallback_for_partially_tuned and fsel.meta.incomplete_tuning:
-                    value = fsel.meta.fallback_tuning_value
-                    fallback_applied.append(fsel)
-                else:
-                    value = fsel.argument_value
-
-                keys[offset] = value
-                if value is None:
-                    assert use_fallback_for_partially_tuned
-                    assert fsel.meta.incomplete_tuning
-                # print(f'keys[{offset}] = {value} {fsel=}')
-            except ValueError:
-                pass
-        l = [keys[offset] for offset in range(len(self._fsel_positions))]
-        # print(f'{l=}')
-        return tuple(l), fallback_applied
+        return True
 
     def _build_db_index(self, fsels):
-        self._init_matching_keys(fsels)
-        self._index = defaultdict(list)
-        self._index_dedup = defaultdict(list)
-        for ti in self._j['tune_info']:
-            tup = self.extract_keys_from_json(ti)
-            # print(f'_build_db_index {tup}')
-            self._index[tup].append(ti)
-            is_dup = False
-            for eti in self._index_dedup[tup]:
-                # print(f'{eti=}')
-                if eti['tuned_kernel'] == ti['tuned_kernel'] and eti['compiler_options'] == ti['compiler_options']:
-                    is_dup = True
-                    break
-            if not is_dup:
-                self._index_dedup[tup].append(ti)
-        if False:  # debug
-            tup=('*fp16:16', 1, 16, True, True)
-            print(f'_build_db_index {self._index[tup]=} {self._index_dedup[tup]=}')
+        assert False
```

Comment:
Isn't this a dead method now?  Can it be removed? Same comment for other functions that are just 'assert False' now.

Reply:
This is an abstract method in the base class and must have some definition.